src/assertions/OrderAssertions.ts

```diff
@@ -37,57 +37,64 @@ import { NULL_ADDRESS } from '../constants';
 import { BigNumber } from '../util';
 import * as Web3 from 'web3';
 
-
-
 export class OrderAssertions {
   private web3: Web3;
+  private coreAddress: Address;
+  private coreContract: CoreContract;
   private erc20Assertions: ERC20Assertions;
   private schemaAssertions: SchemaAssertions;
   private coreAssertions: CoreAssertions;
   private commonAssertions: CommonAssertions;
   private setTokenAssertions: SetTokenAssertions;
 
-  constructor(web3: Web3) {
+  constructor(web3: Web3, coreAddress: Address) {
     this.web3 = web3;
+    this.coreAddress = coreAddress;
     this.erc20Assertions = new ERC20Assertions(web3);
     this.schemaAssertions = new SchemaAssertions();
     this.coreAssertions = new CoreAssertions(web3);
     this.commonAssertions = new CommonAssertions();
     this.setTokenAssertions = new SetTokenAssertions(web3);
   }
 
+  public async ensureCoreContract() {
```

Comment:
Do you still need this? Initialize a core contract in the constructor

Reply:
Wish i could... but the constructor doesn't allow asynchronous methods to get called.

Reply:
Wonder if there's a better solution

Reply:
I see. A factory pattern is the way to go here. You can have an async static method that fetches the core contract and then initializes the OrderAssertions object with it